elasticsearch-evolution-core/src/test/java/com/senacor/elasticsearch/evolution/core/internal/migration/execution/HistoryRepositoryImplTest.java

```diff
@@ -0,0 +1,156 @@
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.rest.RestStatus;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+import org.mockito.Answers;
+import org.mockito.Mock;
+
+import java.io.IOException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Andreas Keefer
+ */
+@ExtendWith(MockitoExtension.class)
+class HistoryRepositoryImplTest {
+
+    public static final String INDEX = "es_evolution";
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private RestHighLevelClient restHighLevelClient;
+    private HistoryRepositoryImpl underTest;
+
+    @BeforeEach
+    void setUp() {
+        underTest = new HistoryRepositoryImpl(restHighLevelClient, INDEX, new MigrationScriptProtocolMapper());
+    }
+
+    @Nested
+    class findAll {
+        @Test
+        void failed() throws IOException {
+            when(restHighLevelClient.search(any(), eq(RequestOptions.DEFAULT)))
+                    .thenThrow(new IOException("test error"));
+
+            assertThatThrownBy(() -> underTest.findAll())
+                    .isInstanceOf(MigrationException.class)
+                    .hasMessage("findAll failed!");
+        }
+    }
+
+    @Nested
+    class saveOrUpdate {
+        @Test
+        void failed() throws IOException {
+            when(restHighLevelClient.index(any(), eq(RequestOptions.DEFAULT)))
+                    .thenThrow(new IOException("test error"));
+            MigrationScriptProtocol protocol = new MigrationScriptProtocol().setVersion("1");
+
+            assertThatThrownBy(() -> underTest.saveOrUpdate(protocol))
+                    .isInstanceOf(MigrationException.class)
+                    .hasMessage("saveOrUpdate of '%s' failed!", protocol);
+        }
+    }
+
+    @Nested
+    class isLocked {
+        @Test
+        void failed() throws IOException {
+            when(restHighLevelClient.indices().refresh(any(), eq(RequestOptions.DEFAULT)).getStatus())
+                    .thenReturn(RestStatus.OK);
+            when(restHighLevelClient.count(any(), eq(RequestOptions.DEFAULT)))
+                    .thenThrow(new IOException("test error"));
+
+            assertThatThrownBy(() -> underTest.isLocked())
+                    .isInstanceOf(MigrationException.class)
+                    .hasMessage("isLocked check failed!");
+        }
+    }
+
+    @Nested
+    class lock {
+        @Test
+        void lock() throws IOException {
```

Comment:
hat das einen grund warum die mehtode nicht failed heißt ?

Reply:
nein, fixed to "failed"